applications/MPMApplication/python_scripts/mpm_analysis.py

```diff
@@ -25,9 +25,37 @@ def __init__(self, model, project_parameters):
             # import KratosMultiphysics.MetisApplication as MetisApplication
             # import KratosMultiphysics.TrilinosApplication as TrilinosApplication
 
+        # add auxiliary variables required by friction automatically to the project_parameters
+        self._AddFrictionAuxiliaryVariables(project_parameters)
+
         super(MPMAnalysis, self).__init__(model, project_parameters)
 
     #### Internal functions ####
+    def _AddFrictionAuxiliaryVariables(self, project_parameters):
+        """ Adds nodal variables required by friction to auxiliary variable list, if needed """
+        aux_var_friction = ["FRICTION_STATE",
+                            "STICK_FORCE"]
+
+        if not project_parameters["processes"].Has("list_other_processes"):
+            return
```

Comment:
I would not do this, these names are intentionally not enforced in other places

Usually, when additional variables are required bcs of some configuration options, they are flags in the solver settings. But this is also not a nice solution, I agree :thinking: 

How about checking every process:
```py
def IsFrictionActive():
    for proc in itertools.chain(project_parameters["processes"].values()):
        if isinstance(proc, ApplyMPMSlipBoundaryProcess):
            if proc_params.Has("friction_coefficient") and proc_params["friction_coefficient"].GetDouble() > 0:
                return True
    return False
```

Pseudo code more or less working, I hope you get the idea

Reply:
Yep, I made the suggested change in the latest commit. `itertools.chain` did not work on the list of parameter objects returned by the `.values()` method so I just used a double for loop instead.